contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6391,6 +6584,691 @@ bbf_ExecDropStmt(DropStmt *stmt)
+	}
+
+	/* Extract the datatype of partitioning column from CREATE statement. */
+	foreach (elements, stmt->tableElts)
+	{
+		Node		*element = lfirst(elements);
+		ColumnDef	*coldef;
+
+		if (nodeTag(element) != T_ColumnDef)
+			continue;
+
+		coldef = castNode(ColumnDef, element);
+
+		if (pg_strcasecmp(coldef->colname, partition_colname) == 0)
+		{
+			Type ctype = LookupTypeName(NULL, coldef->typeName, NULL, true);
+			Form_pg_type pg_type = (Form_pg_type) GETSTRUCT(ctype);
+
+			partition_column_typoid = pg_type->oid;
+			partition_column_basetypoid = pg_type->typbasetype;
+			partition_column_typname = pstrdup(NameStr(pg_type->typname));
+			ReleaseSysCache(ctype);
+			break;
+		}
+	}
+
+	/* Get OID of sql_variant type. */
+	sql_variant_type_oid = (*common_utility_plugin_ptr->get_tsql_datatype_oid) ("sql_variant");
+
+	/*
+	 * Extract metadata of partition function like ranges and input parameter type by
+	 * looking up in bbf_partition_function catalog using the partition function name and dbid.
+	 */
+	rel = table_open(get_bbf_partition_function_oid(), AccessShareLock);
+	
+	ScanKeyInit(&scanKey[0],
+				Anum_bbf_partition_function_dbid,
+				BTEqualStrategyNumber, F_INT2EQ,
+				Int16GetDatum(dbid));
+
+	ScanKeyEntryInitialize(&scanKey[1], 0,
+				Anum_bbf_partition_function_name,
+				BTEqualStrategyNumber, InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_TEXTEQ, CStringGetTextDatum(partition_function_name));
+
+	scan = systable_beginscan(rel, get_bbf_partition_function_pk_idx_oid(),
+					false, NULL, 2, scanKey);
+
+	tuple = systable_getnext(scan);
+
+	if (!HeapTupleIsValid(tuple)) /* Sanity check. */
+	{
+		systable_endscan(scan);
+		table_close(rel, AccessShareLock);
+		ereport(ERROR,
+			(errcode(ERRCODE_UNDEFINED_OBJECT),
+				errmsg("Partition function '%s' used for the specifed partition scheme '%s' does not exist.", partition_function_name, partition_scheme_name)));
+	}
+	
+	input_parameter_type = TextDatumGetCString(heap_getattr(tuple, Anum_bbf_partition_function_input_parameter_type, RelationGetDescr(rel), &isnull));
+	values = DatumGetArrayTypeP(heap_getattr(tuple, Anum_bbf_partition_function_range_values, RelationGetDescr(rel), &isnull));
+	deconstruct_array(values, sql_variant_type_oid, -1, false, 'i', &datum_values, &nulls, &nelems);
+
+	systable_endscan(scan);
+	table_close(rel, AccessShareLock);
+
+	/*
+	 * If the partition columns type is UDT type, then we need
+	 * to use the base type of that type while comparing with
+	 * input parameter type of partition function.
+	 *
+	 */
+	if (OidIsValid(partition_column_basetypoid))
+	{
+		Datum		tsql_typname;
+		/* Get the TSQL typename from partition column type. */
+		InitFunctionCallInfoData(*fcinfo, NULL, 0, InvalidOid, NULL, NULL);
+		fcinfo->args[0].value = ObjectIdGetDatum(partition_column_typoid);
+		fcinfo->args[0].isnull = false;
+		tsql_typname = (*common_utility_plugin_ptr->translate_pg_type_to_tsql) (fcinfo);
```

Comment:
Easier to use (*common_utility_plugin_ptr->get_tsql_datatype_oid) since we already have type name?
And then we can check for invalid oid